rotkehlchen/inquirer.py

```diff
@@ -500,6 +516,161 @@ def find_uniswap_v2_lp_price(
+                len(amount_decoded) != 1 or
+                not isinstance(amount_decoded[0], int)
+            ):
+                log.debug(f'Failed to decode balances {i} while finding curve price. {output}')
+                return None
+            amount = amount_decoded[0]
+            normalized_amount = token_normalized_value_decimals(amount, tokens[i].decimals)
+            data.append(normalized_amount)
+
+        # Prices and data should verifty this relation for the following operations
+        if len(prices) != len(data) - 1:
+            log.debug(
+                f'Lenght of prices {len(prices)} does not match len of data {len(data)} '
+                f'while querying curve pool price.',
+            )
+        # Total number of assets price in the pool
+        total_assets_price = sum(map(operator.mul, data[1:], prices))
+        if total_assets_price == 0:
+            log.error(
+                f'Curve pool price returned unexpected data {data} that lead to a zero price.',
+            )
+            return None
+
+        # Calc weight of each asset as the proportion of tokens value
+        weights = map(lambda x: data[x + 1] * prices[x] / total_assets_price, range(len(tokens)))
+        assets_price = FVal(sum(map(operator.mul, weights, prices)))
+        return (assets_price * FVal(data[0])) / (10 ** lp_token.decimals)
+
+    def find_yearn_price(
+        self,
+        token: EthereumToken,
+    ) -> Optional[Price]:
+        """
+        Query price for a yearn vault v2 token using the pricePerShare method
+        and the price of the underlying token.
+        """
+        assert self._ethereum is not None, 'Inquirer ethereum manager should have been initialized'  # noqa: E501
+
+        maybe_underlying_token = GlobalDBHandler().fetch_underlying_tokens(token.ethereum_address)
+        if maybe_underlying_token is None or len(maybe_underlying_token) != 1:
+            log.error(f'Yearn vault token {token} without an underlying asset')
+            return None
+
+        underlying_token = EthereumToken(maybe_underlying_token[0].address)
+        underlying_token_price = self.find_usd_price(underlying_token)
+        # Get the price per share from the yearn contract
+        contract = EthereumContract(
+            address=token.ethereum_address,
+            abi=YEARN_VAULT_V2_ABI,
+            deployed_block=0,
+        )
+        output = multicall_2(
+            ethereum=self._ethereum,
+            require_success=True,
+            calls=[(token.ethereum_address, contract.encode(method_name='pricePerShare'))],
+        )
+        if isinstance(output, list) and len(output) != 0 and output[0][0] is True:
+            price_per_share = contract.decode(output[0][1], 'pricePerShare')
+            if len(price_per_share) != 0:
+                return price_per_share[0] * underlying_token_price
+            log.error(f'Failed to decode pricePerShare for yearn vault v2 token {token}')
+        log.error(
```

Comment:
If you are not returning None here on purpose and letting it get out of the if then make it obvious to the reader

```suggestion
            log.error(f'Failed to decode pricePerShare for yearn vault v2 token {token}')
            # will return None right below
        log.error(
```

Also as in the previous function, here to perhaps use msg_aggregator to make the user aware? Or is the user seeing something else and they know to check the logs when None is returned?